test/functional/test_runner.py

```diff
@@ -546,6 +551,12 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
                 logging.debug("Early exiting after test failure")
                 break
 
+            shutil_disk_usage_stat_ = shutil.disk_usage(testdir)
+            if (shutil_disk_usage_stat_.free < MIN_FREE_SPACE
+                    and i < test_count - jobs):
+                logging.debug("No space left on device to continue testing")
+                break
```

Comment:
The comment below "    # Clean up dangling processes if any. This may only happen with --failfast option." will need to be updated

Reply:
Yes!
I will let the build finish successfully before changing that.